code/game/machinery/vending.dm

```diff
@@ -1935,6 +1935,59 @@ var/global/num_vending_terminals = 1
 
 	pack = /obj/structure/vendomatpack/engivend
 
+/obj/machinery/vending/building
+	name = "\improper Habitat Depot"
+	desc = "Habitat, sweet habitat. All you need for remodeling."
+	icon_state = "building"
+	products = list(
+		/obj/item/stack/sheet/metal/bigstack = 10,
```

Comment:
I believe it's possible to do this instead: /obj/item/stack/sheet/metal{amount = 20}, removing the need for making a subtype. Also, the /bigstack subtype might have issues with stacking with other stacks